src/transforms/aggregate.rs

```diff
@@ -0,0 +1,367 @@
+        assert_eq!(0, agg.flush_into(&mut out));
+        assert_eq!(0, out.len());
+
+        // Two absolutes with the same series, should get the 2nd (last) back.
+        agg.record(gauge_a_1.clone());
+        agg.record(gauge_a_2.clone());
+        out.clear();
+        assert_eq!(1, agg.flush_into(&mut out));
+        assert_eq!(1, out.len());
+        assert_eq!(&gauge_a_2, &out[0]);
+
+        let gauge_b_1 = make_metric("gauge_b", metric::MetricKind::Absolute,
+            metric::MetricValue::Gauge { value: 44.0 }, tags.clone());
+        // Two increments with the different series, should get each back as-is
+        agg.record(gauge_a_1.clone());
+        agg.record(gauge_b_1.clone());
+        out.clear();
+        assert_eq!(2, agg.flush_into(&mut out));
+        assert_eq!(2, out.len());
+        // B/c we don't know the order they'll come back
+        for event in out {
+            match event.as_metric().series().name.name.as_str() {
+                "gauge_a" => assert_eq!(gauge_a_1, event),
+                "gauge_b" => assert_eq!(gauge_b_1, event),
+                _ => assert!(false),
+            }
+        }
+    }
+
+    #[tokio::test]
+    async fn transform() {
+        let agg = toml::from_str::<AggregateConfig>(
+            r#"
+interval_ms = 10
+"#,
+        )
+        .unwrap()
+        .build(&GlobalOptions::default())
+        .await
+        .unwrap();
+        let agg = agg.into_task();
+
+        let tags: BTreeMap<String, String> =
+            vec![("tag1".into(), "val1".into())].into_iter().collect();
+        let counter_a_1 = make_metric("counter_a", metric::MetricKind::Incremental,
+            metric::MetricValue::Counter { value: 42.0 }, tags.clone());
+        let counter_a_2 = make_metric("counter_a", metric::MetricKind::Incremental,
+            metric::MetricValue::Counter { value: 43.0 }, tags.clone());
+        let counter_a_summed = make_metric("counter_a", metric::MetricKind::Incremental,
+            metric::MetricValue::Counter { value: 85.0 }, tags.clone());
+        let gauge_a_1 = make_metric("gauge_a", metric::MetricKind::Absolute,
+            metric::MetricValue::Gauge { value: 42.0 }, tags.clone());
+        let gauge_a_2 = make_metric("gauge_a", metric::MetricKind::Absolute,
+            metric::MetricValue::Gauge { value: 43.0 }, tags.clone());
+        let inputs = vec![counter_a_1, counter_a_2, gauge_a_1, gauge_a_2.clone()];
+
+        let in_stream = Box::pin(stream::iter(inputs));
+        let mut out_stream = agg.transform(in_stream);
+
+        let mut count = 0;
+        while let Some(event) = out_stream.next().await {
+            count += 1;
+            match event.as_metric().series().name.name.as_str() {
+                "counter_a" => assert_eq!(counter_a_summed, event),
+                "gauge_a" => assert_eq!(gauge_a_2, event),
+                _ => assert!(false),
+            };
+        }
+
+        // There were only 2
+        assert_eq!(2, count);
+    }
```

Comment:
I'd personally feel better if we used a bigger interval, and then used `tokio::time::pause` and `tokio::time::advance` to more concretely demonstrate the "doesn't yield until interval elapses" behavior.

Reply:
Will take a look. Was working off `reduce` as an example and didn't see examples of how to muck with time in tests. Now that I have `tokio::time::pause` to grep for I found a couple examples in other places that I should be able to adapt.

Reply:
I think I addressed this in https://github.com/timberio/vector/pull/7846/commits/1da1e06ba62e0b97200946bfbb3864e1b9416a7d.